spec/bundle/commands/check_command_spec.rb

```diff
@@ -3,13 +3,17 @@
 require "spec_helper"
 
 describe Bundle::Commands::Check do
-  def do_check
-    Bundle::Commands::Check.run
+  RSpec::Matchers.define_negated_matcher :not_raise_error, :raise_error
+
+  def do_check(expected_error = nil, expected_output = //)
```

Comment:
I don't really like this as we're rolling our own version of RSpec's [`raise_error`](https://relishapp.com/rspec/rspec-expectations/v/3-8/docs/built-in-matchers/raise-error-matcher) matcher which seems unnecessary given the tests were previously using explicit assertions that were more concise than this one. Can you provide a little bit of insight into why you've wrapped this up into it's own method versus leaving the tests as they were?

Reply:
Hi @jacobbednarz, thanks for your review.

This is not really rolling our own version of RSpec's `raise_error` - this is using RSpec's built-in support for [defining a negated version](https://relishapp.com/rspec/rspec-expectations/v/3-8/docs/define-negated-matcher) of the existing `raise_error` matcher. This is required, to the best of my knowledge, when defining a [compound expectation](https://relishapp.com/rspec/rspec-expectations/v/3-8/docs/compound-expectations) where one of the clauses is negated (ie some expectation _and_ to _not_ raise an error) as there is [currently no way of doing that](https://github.com/rspec/rspec-expectations/issues/664).

Back to the why - I wanted to verify the output of the command, but currently [this before each](https://github.com/Homebrew/homebrew-bundle/blob/fae70b8e6252f07e0fec170e88738b5ae175827f/spec/bundle/commands/check_command_spec.rb#L12) block stubs all calls to `puts`, presumably to keep the spec output clear.

Given that RSpec has [built-in support for matching output](https://relishapp.com/rspec/rspec-expectations/v/3-8/docs/built-in-matchers/output-matcher) I thought that was a better approach than stubbing `puts` directly, even if I didn't want to verify _what_ the output was, ie just to prevent it being printed to screen.

I could obviously get rid of the logic in `do_check` and inline the compound expectations back into each spec, but I would argue that it's more concise as it is (certainly fewer characters), at the expense of extra complexity in the `do_check` method.

Personally I would favour the current approach in terms of being more concise, but happy to switch to inlining it if you'd prefer, being simpler at the expense of more verbose.

Anyway, let me know your thoughts.

Reply:
We don't need to check for an exception not being raised because the test will fail if the exception is raised.